lib/studies/reply/adaptive_nav.dart

```diff
@@ -131,6 +134,20 @@ class _DesktopNavState extends State<_DesktopNav>
           });
         }
       });
+
+    final destinationKeysList = widget.destinations.keys.toList();
+
+    _indexWithDestinations = {
+      for (var i = 0; i < widget.destinations.keys.length; i++)
+        i: destinationKeysList[i]
+    };
+
+    destinationKeysList.clear();
+
+    _currentInbox = InboxPage(
+      key: UniqueKey(),
+      destination: _indexWithDestinations[widget.selectedIndex],
+    );
```

Comment:
1. You have very similar logic in the mobile and desktop NavStates. This is a clue it could be consolidated in the parent `_AdaptiveNavState`. You could then just pass down the _currentInbox Widget.

2. I think `_BottomDrawerDestinations` could be greatly simplified. What if instead of identifying destinations by `String`, you had a `Destination` class containing the index and name. You could then just pass down a `List<Destination>` and `currentlySelectedIndex`.
`destinationsWithIndex[destination] == selectedIndex` would become something like `destination.index == selectedIndex`. I would try this first before 1.

Reply:
Thanks for the great feedback! I'll explore 2.

Reply:
Both 1 and 2 are now done!

Reply:
Great! It's even cleaner than I imagined